core/src/html/layout.rs

```diff
@@ -655,7 +676,7 @@ impl<'a, 'gc> LayoutContext<'a, 'gc> {
             "_明朝" => Some(DefaultFont::JapaneseMincho),
             _ => None,
         } {
-            return context
+            if let Some(&font) = context
```

Comment:
Did this happen? I am not sure we really want to even support a missing default font. (I guess this could happen on custom Linux installs, without Noto Sans)

Reply:
Ah. I see, on the web with a custom config.

Maybe upgrade this to error, and immediately return an empty font?

Reply:
This has also happened on a Windows desktop: https://github.com/ruffle-rs/ruffle/issues/18816 :/

We also have to "support" missing fonts at least by not panicking :smile: We also shouldn't skip any text or text layout when a font is missing and should assume metrics of value 0 (FP does that). So making `resolve_font` always resolve to something seems very appropriate.

I was wondering whether to error out here or fall back, but I thought it would be best to do everything to display _something_, especially that (according to my knowledge) we don't have a lot of tests related to device font resolution and don't match FP 100% in this regard.

But I personally don't have that much experience with fonts here, so if you think it's best to error out I'll happily change that, just let me know.

Reply:
I think it would make more sense to:
- In `default_font` try harder to find any available font? i.e. `device_fonts.all()`
- Otherwise return the empty font and log an error (So we don't keep creating empty fonts / logging errors)
- Revert the changes to this if branch